hazelcast/src/main/java/com/hazelcast/map/impl/MapKeyLoaderUtil.java

```diff
@@ -86,15 +86,14 @@ public boolean hasNext() {
     }
 
     public static int getMaxSizePerNode(MaxSizeConfig maxSizeConfig) {
-
-        int maxSizePerNode = getApproximateMaxSize(maxSizeConfig, MaxSizePolicy.PER_NODE);
+        double maxSizePerNode = getApproximateMaxSize(maxSizeConfig, MaxSizePolicy.PER_NODE);
 
         if (maxSizePerNode == MaxSizeConfig.DEFAULT_MAX_SIZE) {
             // unlimited
             return -1;
         }
 
-        return maxSizePerNode;
+        return (int) maxSizePerNode;
```

Comment:
That narrowing should be fine, at some point we should do that. Main issue we are trying to fix here appears in percentage comparisons.